### ISSUE TITLE
Refactor fix for #4

### ISSUE THREAD

**Issue by clayrichardson** [absortium/poloniex-api#7]
It might make more sense to refactor, and pass in the whole WAMP event to the handler, and let each handler deal with the message fields themselves.

**Comment by AndrewSamokhvalov** [comment]
@clayrichardson Yeah, at the time when I used this library my goal was just to make it work :) Actually most of the `wamp` code have to be rewritten, but currently I am occupied with another project. It would be very pleasant for me to see if somebody took the responsibility to rewrite some of this code.
